linode/manager
# feat: [M3-7120] - Create VPC Feedback

## Description 📝
Address VPC Create page related feedback from product

## Major Changes 🔄
- scroll into error for general subnet errors
- better readability for available IPs #
- hold off on displaying # of available IPs for non RFC1918 IPs

## Preview 📷

Commas for large numbers
| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/6fb24d7d-d563-491c-84d6-c56c05275847) | ![image](https://github.com/linode/manager/assets/139280159/7a52c6c4-0602-4cb1-8386-363592599253) |

Number of available ips doesn't show up for a non RFC1918 ip 
| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/6e873556-2ffb-4be3-9fe6-c47583a60a5d) | ![image](https://github.com/linode/manager/assets/139280159/65dd015a-d5f1-46f4-8a4b-da4d374000ae) |

scroll into view subnet errors
https://github.com/linode/manager/assets/139280159/e3d5218a-2f53-4bc6-8472-7bd08d6a6c34

## How to test 🧪
**How to verify changes?**
You can use the MSW (with VPC feature flag) for the following:
- On the Create VPC page (as well as the Create Subnet drawer on a VPC's detail page), confirm the following:
  - The # of available IPs message will only show up for RFC1918 IPs.
    - For 10.x.x.x IPs, the mask must be >= 8
    - For 172.16.x.x-172.31.x.x, the mask must be >= 12
    - For 192.168.x.x, the mask must be >= 16
  - If a number shows up and is larger than 999, commas appear

You will need to be on the dev environment with VPC tags on your account for the following:
- Navigate to the VPC Create page. Verify that if general subnet errors pop up, the page scrolls to them.
  - To produce some of these errors: Input a valid region (currently I believe only Newark works) and VPC label. Then, in the subnets section, add 3-4+ subnets (enough that you can no longer see the first subnet) and give them all the same label and IP. Click `Create VPC`, and you should see that your screen scrolls to the errors that have appeared. 

 **How to run Unit or E2E tests?**
`yarn test subnets.test.ts`
